lib/subscription.coffee

```diff
@@ -12,14 +12,19 @@ class Subscription extends EventEmitter
 
     @stream.on 'data', (data) =>
       @connected = true
-      if @options.json
-        str = data.toString()
 
-        if !str then return
-
-        data = JSON.parse str
-
-      @emit 'data', data
+      # It was reported that at times there can be multiple messages in the buffer
+      # somehow, so to be safe the data string is split and processed for each message
+      messages = switch
+        when @options.json
+          data
+            .toString()
+            .split /\r?\n/
+            .filter Boolean
```

Comment:
Note though wouldn't catch more than one space lol... I don't think that is worth covering though yet as this is really for catching stuff like empty array to string etc.

Also there is no check for 'type' anywhere so... not sure what kind of text in the buffer is being streamed that would just have new lines and no valid json... maybe there is some kind of maintenance message or something Oanda are sending back which isnt an error response?